dbt/adapters/databricks/impl.py

```diff
@@ -699,12 +699,14 @@ def get_persist_doc_columns(
         # an error when we tried to alter the table.
         for column in existing_columns:
             name = column.column
-            if (
-                name in columns
-                and "description" in columns[name]
-                and columns[name]["description"] != (column.comment or "")
-            ):
-                return_columns[name] = columns[name]
+            if name in columns:
+                config_column = columns[name]
+                if isinstance(config_column, dict):
+                    comment = columns[name].get("description", "")
+                else:
+                    comment = config_column.description
```

Comment:
do you want to check with `hasattr` before accessing it?

Reply:
Good call, that will be safer.